sdk/cosmosdb/cosmos/src/queryExecutionContext/EndpointComponent/OffsetLimitEndpointComponent.ts

```diff
@@ -22,18 +25,55 @@ export class OffsetLimitEndpointComponent implements ExecutionContext {
     const buffer: any[] = [];
     const response = await this.executionContext.fetchMore(diagnosticNode);
     mergeHeaders(aggregateHeaders, response.headers);
-    if (response === undefined || response.result === undefined) {
-      return { result: undefined, headers: response.headers };
+    if (
+      response === undefined ||
+      response.result === undefined ||
+      !Array.isArray(response.result.buffer) ||
```

Comment:
In what case this will be be non array?

Reply:
it can be a falsie (undefined, null,) value.